### PR TITLE
Add view to get timeline

### PR DESCRIPTION
**Pull request by ArianaGu**
Also alter the post model to allow users to 'like' a post

### CODE REVIEW

File: backend/graminsta/post/models.py [hunk 1]
```diff
@@ -17,10 +17,17 @@ class Post(models.Model):
...
+                                         blank=True)
+    mention_user = models.ManyToManyField(auth_models.User,
+                                          related_name="mentioned_by_post",
+                                          blank=True)
     created_at = models.DateTimeField(auto_now_add=True)
+    kudos = models.IntegerField(default=0)
+
+    def __str__(self):
+        return '%s: %s' % (self.publisher.username,
+                           self.description)
```
Comment (Time1ess) [inline backend/graminsta/post/models.py]: I think you can start with:

1. My posts
2. Posts from my following people

So you don't need to make these model changes.

File: backend/graminsta/post/services.py [hunk 2]
```diff
@@ -90,3 +91,17 @@ def delete_follow_relationship(request_user, target_user):
...
+def get_timeline_posts(request_user):
+    """
+    Returns posts that should be displayed at timeline, including
+    posts created by the request user and his/her following people,
+    and posts which mention or mark the request user.
+    """
+    following_people = get_people_user_follows(request_user)
+    posts = Post.objects.filter(Q(publisher__in=following_people) |
+                                Q(publisher=request_user) |
+                                Q(mention_user=request_user) |
```
Comment (Time1ess) [inline backend/graminsta/post/services.py:105]: Can remove this for a simple start

**Review by Time1ess — CHANGES_REQUESTED** [review]

File: backend/graminsta/post/models.py (hunk 1 above)
Comment (ArianaGu) [reply backend/graminsta/post/models.py]: Actually I only add `blank=True` feature to make `mention_user` and `marked_user` field not required, these two fields did exist before in dev branch, do I need to remove them?

Comment (ArianaGu) [reply backend/graminsta/post/models.py]: [This PR](https://github.com/SPS-2020-Team-15/Graminsta/pull/11) adds `marked_user` and `mention_user` field, I think these two fields should not be required when I created a post object to test, so I add the `blank=True` feature.

Comment (Time1ess) [reply backend/graminsta/post/models.py]: Got it, but in order to make this field optional, you need to add `null=True` too. `blank=True` only bypass Django field validation, whereas `null=True` allows the field to be null in database level

**Review by Time1ess — APPROVED** [review]

**Review by Time1ess — APPROVED** [review]
Please help re-generate the migration file

**Review by shiyi001 — APPROVED** [review]

**Pull request merged**